### ISSUE TITLE
[META] Algorithms to implement

### ISSUE THREAD

**Issue by AKuederle** [mobilise-d/mobgap#8]
![Image](https://user-images.githubusercontent.com/10776763/264980104-da250b43-a6cc-4357-b987-3d72aabb6ba1.png)

The Picture above shows the full pipeline we need to implement. For most algorithmic steps this means one algorithm. Only for GSD and CAD there are two.

In addition to the blocks shown above, there is a Reorientation Block that should catch instances where the signal is wrongly oriented.

## Overview

- [ ] Reorientation block (#112 )
- [x] GSD 1 (#21)
- [x] GSD 2 (#24)
- [x] GSD 3 ("the real GSD 2") (#143 )
- [x] Turning (#31)
- [x] IC-Detection (#28 )
- [x] CAD 1/IC 2 (#29 #72 )
- [x] CAD 2/IC 3 (#29 #84 )
- [x] Cad2Sec (#61)
- [x] SL (#142  )
- [x] Left Right - Normal (#32, #100 )
- [x] Left Right - ML pretrained (#32)
- [x] Walking Speed (#148 )
- [x] Stride interpolation (#141 )
- [x] WBA (#35 )
- [x] Dmo cleaning (#12, #91 )
- [x] Aggregation Methods (#13)

## How to implement new algorithms

**When implementing a new algorithm, create a new issue with the [template](https://github.com/mobilise-d/gaitlink/issues/new?assignees=&labels=New+Algorithm&projects=&template=new-algorithm.md&title=%5BALGO%5D+%7BAlgorithm+Name%7D) and update this issue with a link to all relevant issues and PRs.**

1. Find the old implementation and try to understand it. What is the actual core algorithm? What is just pre-processing? What are the relevant parameters of the algorithm? What is the output type (per-second, per-step)? What inputs are required (raw data, sensor height, detected steps, ...)?
2. Get the old version to run offline on the example data. "Mock" previous outputs using the ground truth. Store the results somewhere
3. Discuss in the group, before starting any implementation. Are the outputs as expected? Do we understand how the algorithm is supposed to work?
4. Start building an Example (i.e. a notebook or Python file to prototype). Have your raw inputs and your example output from before there to play around.
5. Create a first implementation of the core functionality. Just build a simple Python function. Ignore class interface for now (I would suggest to try AI-tools to make the initial conversion from matlab to Python). Check if your implementation gets the same output as before. **If there are deviations, that can not/should not be "fixed" document them**
6. Build the class interface + tests + docu + example. Example should cover basic usage of algorithm + comparison to old matlab output. Write down anything that could interesting for someone in the future.
